lib/Sema/CSSimplify.cpp

```diff
@@ -9513,14 +9517,47 @@ performMemberLookup(ConstraintKind constraintKind, DeclNameRef memberName,
   if (isSingleUnlabeledPackExpansionTuple(instanceTy)) {
     auto elementTy = instanceTy->castTo<TupleType>()->getElementType(0);
     if (elementTy->is<TypeVariableType>()) {
-      MemberLookupResult result;
       result.OverallResult = MemberLookupResult::Unsolved;
       return result;
     }
   }
 
+  // Delay solving member constraint for unapplied methods
+  // where the base type has a conditional Sendable conformance
+  if (ctx.LangOpts.hasFeature(Feature::InferSendableMethods)) {
+    if (functionRefKind == FunctionRefKind::Unapplied) {
+      auto sendableProtocol =
+          DC->getParentModule()->getASTContext().getProtocol(
+              KnownProtocolKind::Sendable);
+      auto baseConformance = DC->getParentModule()->lookupConformance(
+          instanceTy, sendableProtocol);
+
+      if (llvm::any_of(
+              baseConformance.getConditionalRequirements(),
+              [&](const auto &req) {
+                switch (req.getKind()) {
```

Comment:
So in this `switch` there is only one actionable case, why not simplify it down to if:

```c++
if (req.getKind() != RequirementKind::Conformance)
  return false;

if (auto protocolTy = req.getSecondType()->template getAs<ProtocolType>()) {
  return req.getFirstType()->hasTypeVariable() &&
         protocolTy->getDecl()->isSpecificProtocol(KnownProtocolKind::Sendable);
}
return false;
```